cmake_modules/FindMYSQL.cmake

```diff
@@ -46,12 +46,16 @@ IF (NOT MYSQL_FOUND)
       SET (osdir "unknown")
     ENDIF()
     IF (NOT ("${osdir}" STREQUAL "unknown"))
-      FIND_PATH (MYSQL_INCLUDE_DIR NAMES mysql.h PATHS "${EXTERNALS_DIRECTORY}/mysql/${osdir}/include"
-        "${EXTERNALS_DIRECTORY}/mysql/include" NO_DEFAULT_PATH)
-      FIND_LIBRARY (MYSQL_LIBRARIES NAMES ${mysql_lib} PATHS "${EXTERNALS_DIRECTORY}/mysql/${osdir}/lib"
-        "${EXTERNALS_DIRECTORY}/mysql/${osdir}" "${EXTERNALS_DIRECTORY}/mysql/lib" NO_DEFAULT_PATH)
-      FIND_LIBRARY (MYSQL_LIBRARIES NAMES ${mysql_lib_alt} PATHS "${EXTERNALS_DIRECTORY}/mysql/${osdir}/lib"
-        "${EXTERNALS_DIRECTORY}/mysql/${osdir}" "${EXTERNALS_DIRECTORY}/mysql/lib" NO_DEFAULT_PATH)
+      IF( "${MYSQL_INCLUDE_DIR}" STREQUAL "" )
+        FIND_PATH (MYSQL_INCLUDE_DIR NAMES mysql.h PATHS "${EXTERNALS_DIRECTORY}/mysql/${osdir}/include"
+          "${EXTERNALS_DIRECTORY}/mysql/include" NO_DEFAULT_PATH)
+      ENDIF()
+      IF( "${MYSQL_LIBRARIES}" STREQUAL "" )
```

Comment:
I wonder if it ever makes sense to specify MYSQL_INCLUDE_DIR and not MYSQL_LIBRARIES or the opposite ?
Maybe a single check for both MYSQL_INCLUDE_DIR and MYSQL_LIBRARIES being set ?
Can you add the actual values you use when building el7 as a comment somewhere so its documented ?
(maybe values used could be in a different JIRA/PR, but it would be really nice to be able to see the cmake command-line options in code somewhere)

Reply:
But otherwise the change to only call FIND_PATH and/or FIND_LIBRARY when these are not set looks ok.

Reply:
Here is cmake option example,

     -DMYSQL_INCLUDE_DIR=/usr/local/mysql/include -DMYSQL_LIBRARIES=/usr/local/mysql/lib/libmysqlclient.a \

All examples are in http://10.224.20.18/view/MySQL/